pytorch_lightning/utilities/types.py

```diff
@@ -44,6 +44,7 @@
     Dict[str, Sequence[DataLoader]],
 ]
 EVAL_DATALOADERS = Union[DataLoader, Sequence[DataLoader]]
+_DEVICE = Union[torch.device, str, int]
```

Comment:
In which case the devices will be int? I feel this losses typing restrict for some classes/method. For example, in GPUAccelerator, before the device has to be torch.device or str, now int will pass the type check too.

Reply:
`pytorch` defines `int` as a valid `device` in their functions:

https://github.com/pytorch/pytorch/blob/62441157e3b5ee96e9cb99f7621422b8bf394d09/torch/cuda/memory.py#L117

Reply:
We can either add support for our device or not align with `pytorch` at this time, should we have an issue support int device across our code base?

Reply:
Right now this only impacts `SingleDeviceStrategy`, hence why I added

`self._root_device = torch.device(device)`

as `self.root_device` is always a `torch.device`.

I don't think `self.root_device` needs to support `str` or `int`.

Reply:
Sure, we can discuss if this type been used somewhere else or this topic come up again